src/Expressions.ts

```diff
@@ -0,0 +1,202 @@
+  strLength: 'length',
+  strMatches: null,
+  strReplace: null,
+  strStartsWith: null,
+  strStripAccents: null,
+  strSubstring: 'slice',
+  strSubstringStart: null,
+  strToLowerCase: 'downcase',
+  strToUpperCase: 'upcase',
+  strTrim: null,
+  // ---- number ----
+  add: '+',
+  abs: 'abs',
+  acos: 'acos',
+  asin: 'asin',
+  atan: 'atan',
+  atan2: null,
+  ceil: 'ceil',
+  cos: 'cos',
+  div: '/',
+  exp: 'e',
+  floor: 'floor',
+  log: 'ln',
+  // – : 'ln2'
+  // – : 'log10'
+  // – : 'log2'
+  max: 'max',
+  min: 'min',
+  modulo: '%',
+  mul: '*',
+  pi: 'pi',
+  // - : 'e',
+  pow: '^',
+  random: null,
+  rint: null,
+  round: 'round',
+  sin: 'sin',
+  sqrt: 'sqrt',
+  sub: '-',
+  tan: 'tan',
+  toDegrees: null,
+  toRadians: null,
+  // ---- boolean ----
+  all: 'all',
+  // eslint-disable-next-line id-blacklist
+  any: 'any',
+  between: 'within',
+  double2bool: null,
+  equalTo: '==',
+  greaterThan: '>',
+  greaterThanOrEqualTo: '>=',
+  in: 'in',
+  lessThan: '<',
+  lessThanOrEqualTo: '<=',
+  not: '!',
+  notEqualTo: '!=',
+  parseBoolean: 'to-boolean',
+  // ---- unknown ----
+  case: 'case',
+  property: 'get'
+};
+
+const invertedFunctionNameMap: Partial<Record<ExpressionName, GeoStylerFunction['name']>> =
+  invert(functionNameMap);
+
+export function gs2mbExpression<T extends PropertyType>(gsExpression?: GeoStylerExpression<T>):
+  MapboxExpression | T | undefined {
+  if (!isGeoStylerFunction(gsExpression)) {
+    return gsExpression as T;
+  }
+
+  const args = 'args' in gsExpression ? gsExpression.args : [];
+
+  // special handling
+  switch (gsExpression.name) {
+    case 'case':
+      const mbArgs: any = [];
+      args.forEach((arg: any, index: number) => {
+        if (index === (args.length - 1)) {
+          mbArgs.push(gs2mbExpression(arg));
+        } else if (arg.case && arg.value) {
+          mbArgs.push(gs2mbExpression(arg.case));
+          mbArgs.push(gs2mbExpression(arg.value));
+        } else {
+          throw new Error('Could not translate GeoStyler Expression: ' + gs2mbExpression);
+        }
+      });
+      return ['case', ...mbArgs];
+    case 'exp':
+      if (args[0] === 1) {
+        return ['e'];
+      }
+      break;
+    case 'pi':
+      return ['pi'];
+    default:
+      const mapboxFunctionName = functionNameMap[gsExpression.name];
+      if (!mapboxFunctionName) {
+        throw new Error('Could not translate GeoStyler Expression: ' + gsExpression);
+      }
+      return [mapboxFunctionName, ...args.map(arg => gs2mbExpression(arg))] as MapboxExpression;
+  }
+
+  throw new Error('Could not translate GeoStyler Expression: ' + gsExpression);
+
+}
+
+type MbInput = MapboxExpression | PropertyType | StyleFunction;
+
+export function mb2gsExpression<T extends PropertyType>(mbExpression?: MbInput):
+  GeoStylerExpression<T> | undefined {
+
+  // TODO: is this check valid ?
+  // fails for arrays like offset = [10, 20]
+  if (!(Array.isArray(mbExpression) && mbExpression[0])) {
```

Comment:
I don't think `&& mbExpression[0]` is really helpful. Expressions can be identified by arrays that have a string as the first value. That string could also be checked with a lookup table (something like a complete version of `invertedFunctionNameMap`) but that might be future work.

For now, I think checking for the first item being a string is already an improvement.